lib/index.js

```diff
@@ -0,0 +1,322 @@
+// FIXME: should joyent copyright be included here `parseRequest` is mostly
+// copy/paste
+
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const jsprim = require('jsprim');
+const sprintf = require('util').format;
+const url = require('url');
+const HttpSignatureError = require('./HttpSignatureError');
+
+const api = {};
+
+module.exports = api;
+
+api.createAuthzHeader = ({algorithm, includeHeaders, keyId, signature}) => {
+  assert.optionalString(algorithm);
+  assert.string(keyId);
+  assert.arrayOfString(includeHeaders);
+  assert.string(signature);
+
+  let authzFormat;
+  let authz;
+  if(algorithm) {
+    authzFormat =
+      'Signature keyId="%s",algorithm="%s",headers="%s",signature="%s"';
+    authz = sprintf(
+      authzFormat, keyId, algorithm, includeHeaders.join(' '), signature);
+  } else {
+    authzFormat = 'Signature keyId="%s",headers="%s",signature="%s"';
+    authz = sprintf(
+      authzFormat, keyId, includeHeaders.join(' '), signature);
+  }
+  return authz;
+};
+
+api.createStringToSign = ({includeHeaders, requestOptions} = {}) => {
```

Comment:
Let's call this `createSignatureString` to match "Signature String Construction" in the spec. This also works better when you're creating it to verify, not sign.